WordPress/gutenberg
# Embed Block: Border support added

## What?

Adding border support for Embed block.
Part of:  https://github.com/WordPress/gutenberg/issues/43247

## Why?

Embed block is missing border support


## How?

Adds the border support via block.json


## Testing Instructions

- Go to the Global Styles setting.
- Make sure that the Embed block's border is configurable via Global Styles.
- Verify that Global Styles are applied correctly in the editor and frontend.
- Navigate to post/page edit.
- Then add Embed block and apply the border styles.
- Verify that block border styles take precedence over global block colour settings.
- Verify that block borders display correctly in both the editor and frontend.
- Border will get applied only to the embeds, not to captions.

NOTE:

1. WP embeds are handling in bit different way, so make sure all embeds are working well.  You can try with all possible embeds available like YouTube, X(Twitter), Spotify, SoundCloud and WordPress posts.
2. This PR does not handle border-radius since it requires padding support as well. Adding padding support would involve reconsidering several scenarios, which could complicate the implementation. To keep this PR straightforward, border-radius and padding are not addressed at this time.


### Sample URLs for Testing:

WordPress Post:
https://wordpress.org/documentation/article/embeds/

YouTube Video:
https://www.youtube.com/watch?v=sYNqeahE2rg

Tweet:
https://x.com/WordPress

Spotify Playlist:
https://open.spotify.com/playlist/37i9dQZF1DXcBWIGoYBM5M

SoundCloud Track:
https://soundcloud.com/forss/flickermood

## Screenshots or screencast 


https://github.com/user-attachments/assets/53186fed-812b-4fe4-9b14-57a45745d3e4
